builtin/mainmenu/dlg_create_world.lua

```diff
@@ -377,36 +384,48 @@ local function create_world_buttonhandler(this, fields)
 				worldname = "world" .. worldnum_max + 1
 			end
 
-			core.settings:set("fixed_map_seed", fields["te_seed"])
-
-			local message
-			if not menudata.worldlist:uid_exists_raw(worldname) then
-				core.settings:set("mg_name",fields["dd_mapgen"])
-				message = core.create_world(worldname,gameindex)
-			else
+			if menudata.worldlist:uid_exists_raw(worldname) then
 				message = fgettext("A world named \"$1\" already exists", worldname)
 			end
+		end
 
-			if message ~= nil then
-				gamedata.errormessage = message
-			else
-				core.settings:set("menu_last_game",pkgmgr.games[gameindex].id)
-				if this.data.update_worldlist_filter then
-					menudata.worldlist:set_filtercriteria(pkgmgr.games[gameindex].id)
-					mm_game_theme.update("singleplayer", pkgmgr.games[gameindex].id)
-				end
-				menudata.worldlist:refresh()
-				core.settings:set("mainmenu_last_selected_world",
-									menudata.worldlist:raw_index_by_uid(worldname))
+		if message == nil then
+			this.data.seed = fields["te_seed"]
+			this.data.mg = fields["dd_mapgen"]
+
+			-- actual names as used by engine
+			local settings = {
+				fixed_map_seed = this.data.seed,
+				mg_name = this.data.mg,
+				mg_flags = table_to_flags(this.data.flags.main),
+				mgv5_spflags = table_to_flags(this.data.flags.v5),
+				mgv6_spflags = table_to_flags(this.data.flags.v6),
+				mgv7_spflags = table_to_flags(this.data.flags.v7),
+				mgfractal_spflags = table_to_flags(this.data.flags.fractal),
+				mgcarpathian_spflags = table_to_flags(this.data.flags.carpathian),
+				mgvalleys_spflags = table_to_flags(this.data.flags.valleys),
+				mgflat_spflags = table_to_flags(this.data.flags.flat),
+			}
+			message = core.create_world(worldname, gameindex, settings)
+		end
+
+		if message == nil then
+			core.settings:set("menu_last_game", game.id)
+			if this.data.update_worldlist_filter then
+				menudata.worldlist:set_filtercriteria(game.id)
```

Comment:
Is this filter also applied intentionally for Android? After the world creation, the filter cannot be undone.

Reply:
IIRC the `update_worldlist_filter` option didn't work before so changing this just revealed an old "bug"
Will change